libraries/script-engine/src/ScriptEngine.h

```diff
@@ -118,6 +118,17 @@ class EntityScriptDetails {
  *       <li><code>"agent"</code>: An assignment client script.</li>
  *     </ul>
  *     <em>Read-only.</em>
+ * @property {string} type - The type of engine that is currently in use:
+ *     <ul>
+ *       <li><code>"client"</code>: The engine is an interface.</li>
+ *       <li><code>"entity_client"</code>: The engine is an entity simulated in an interface.</li>
+ *       <li><code>"entity_server"</code>: The engine is an entity simulated in a server.</li>
+ *       <li><code>"agent"</code>: The engine is an agent.</li>
+ *       <li><code>"avatar"</code>: The engine is an avatar.</li>
+ *     </ul>
```

Comment:
Hmmm. These are a little different from the script types listed at the top of the JSDoc pages: Interface Scripts • Client Entity Scripts • Avatar Scripts • Server Entity Scripts • Assignment Client Scripts ?
If so, add a note that they are different (and perhaps why / how)?

Reply:
If I knew why... :D

Reply:
This originated in the modkit feature, which distinguishes between Interface and avatar script "contexts", whereas the "context" property does not.